src/main/java/com/google/googleidentity/oauth2/endpoint/TokenRevokeEndpoint.java

```diff
@@ -0,0 +1,158 @@
+    this.oauth2TokenService = oauth2TokenService;
+  }
+
+  protected void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException, UnsupportedOperationException {
+    OAuth2Exception exception =
+        new InvalidRequestException(InvalidRequestException.ErrorCode.UNSUPPORTED_REQUEST_METHOD);
+    log.info("Token Revoke endpoint does not support GET request.");
+    OAuth2ExceptionHandler.handle(exception, response);
+    return;
+  }
+
+  protected void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException, UnsupportedOperationException {
+
+    try {
+      TokenRevokeEndpointRequestValidator.validatePOST(request);
+    } catch (OAuth2Exception exception) {
+      log.info(
+          "Failed in validating Post request in Token Revoke Endpoint."
+              + "Error Type: "
+              + exception.getErrorType()
+              + "Description: "
+              + exception.getErrorDescription());
+      OAuth2ExceptionHandler.handle(exception, response);
+      return;
+    }
+
+    Preconditions.checkArgument(
+        OAuth2Utils.getClientSession(request).getClient().isPresent(),
+        "Client should have been set in client filter!");
+    ClientDetails client = OAuth2Utils.getClientSession(request).getClient().get();
+
+    String tokenTypeHint = request.getParameter("token_type_hint");
+
+    if (tokenTypeHint == null) {
+      tokenTypeHint = TokenTypes.ACCESS_TOKEN;
+    }
+    OAuth2Request.Builder builder = OAuth2Request.newBuilder();
+    builder.getRequestAuthBuilder().setClientId(client.getClientId());
+    builder
+        .getRequestBodyBuilder()
+        .setTokenToRevoke(request.getParameter("token"))
+        .setTokenTypeHint(OAuth2EnumMap.TOKEN_TYPE_MAP.get(tokenTypeHint));
+    try {
+      revokeToken(response, builder.build());
+    } catch (OAuth2Exception exception) {
+      log.info(
+          "Failed when process request in Token Revoke Endpoint"
+              + "Error Type: "
+              + exception.getErrorType()
+              + "Description: "
+              + exception.getErrorDescription());
+      OAuth2ExceptionHandler.handle(exception, response);
+      return;
+    }
+    JSONObject json = new JSONObject();
+    response.setContentType("application/json;charset=UTF-8");
+    response.getWriter().println(json.toJSONString());
+    response.getWriter().flush();
+  }
+
+  @VisibleForTesting
+  void revokeToken(HttpServletResponse response, OAuth2Request request)
+      throws InvalidGrantException {
+    String token = request.getRequestBody().getTokenToRevoke();
+
+    switch (request.getRequestBody().getTokenTypeHint()) {
+      case ACCESS:
+        Optional<OAuth2AccessToken> oldAccessToken = oauth2TokenService.readAccessToken(token);
+        if (!oldAccessToken.isPresent()) {
+          throw new InvalidGrantException(ErrorCode.NONEXISTENT_REVOKE_TOKEN);
```

Comment:
Should this be considered success when the token to be revoked does not exist? Is there any document about this case?